network-api/networkapi/wagtailpages/pagemodels/base.py

```diff
@@ -438,6 +439,59 @@ class ParticipateHighlights2(ParticipateHighlightsBase):
     )
 
 
+@register_snippet
+class FocusArea(models.Model):
+    interest_icon = models.ForeignKey(
+        'wagtailimages.Image',
+        null=True,
+        on_delete=models.SET_NULL,
+        related_name='interest_icon'
+    )
+
+    name = models.CharField(
+        max_length=100,
+        help_text='The name of this area of focus. Max. 100 characters.',
+    )
+
+    description = models.TextField(
+        max_length=300,
+        help_text='Description of this area of focus. Max. 300 characters.',
+    )
+
+    url = models.URLField(
+        max_length=2048,
+        help_text='URL for this area of focus\' dedicated page.',
+    )
+
+    panels = [
+        ImageChooserPanel('interest_icon'),
+        FieldPanel('name'),
+        FieldPanel('description'),
+        FieldPanel('url'),
```

Comment:
@KalobTaulien do  you know if there is a way to make this use the same kind of URL dialog that we get in the RichText editor?

Reply:
Yep, but it's a two pronged approach when using standard model fields. 

```
internal_url = models.ForeignKey('wagtailcore.Page', ..., blank=True)
external_url = models.URLField(..., blank=True) 

@property 
def url(self):
    if self.internal_url:
        return self.internal_url.url 
    return self.external_url 

panels = [
    ....
    PageChooserPanel("internal_url"),
    FieldPanel("external_url")
]
```

Note I didnt check that for typos, but that's the gist of it. We can also add page validation to make sure both fields are not set and help_text to help editors know what to do.

Reply:
I'm going to mark this as resolved because we explored a way to do this already.